src/core/Value.h

```diff
@@ -461,3 +464,36 @@ std::ostream& operator<<(std::ostream& stream, const Value::ObjectType& u);
 using VectorType = Value::VectorType;
 using EmbeddedVectorType = Value::EmbeddedVectorType;
 using ObjectType = Value::ObjectType;
+
+class VirtualValueIterator
```

Comment:
When working with iterator-like stuff, you want to define `difference_type`, `value_type`, `pointer`, `reference` and `iterator_category` (see https://en.cppreference.com/w/cpp/iterator/iterator_traits.html). It is a bit annoying but worth the hassle.